components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/utils/ThumbnailDiskCache.kt

```diff
@@ -22,13 +23,19 @@ private const val WEBP_QUALITY = 90
  */
 class ThumbnailDiskCache {
     private val logger = Logger("ThumbnailDiskCache")
-    private var thumbnailCache: DiskLruCache? = null
+    @VisibleForTesting
+    internal var thumbnailCache: DiskLruCache? = null
     private val thumbnailCacheWriteLock = Any()
 
     internal fun clear(context: Context) {
         synchronized(thumbnailCacheWriteLock) {
-            getThumbnailCache(context).delete()
-            thumbnailCache = null
+            try {
+                getThumbnailCache(context).delete()
+            } catch (e: IOException) {
+                logger.warn("Thumbnail cache could not be cleared. Perhaps there are none?")
+            } finally {
+                thumbnailCache = null
+            }
```

Comment:
nit: Technically you do not need to put this into a `finally` block here :)

Reply:
Thanks!

I was curious, so I looked at the generated Java code. This is what it looks like with the `finally` block:
```java
synchronized(var2){}

try {
   boolean var5 = false;

   try {
      this.getThumbnailCache(context).delete();
   } catch (IOException var13) {
      Logger.warn$default(this.logger, "Thumbnail cache could not be cleared. Perhaps there are none?", (Throwable)null, 2, (Object)null);
   } finally {
      this.thumbnailCache = (DiskLruCache)null;
   }

   Unit var16 = Unit.INSTANCE;
} finally {
   ;
}
```

And here it is without the `finally`:
```java
synchronized(var2) {
   boolean var5 = false;

   try {
      this.getThumbnailCache(context).delete();
   } catch (IOException var7) {
      Logger.warn$default(this.logger, "Thumbnail cache could not be cleared. Perhaps there are none?", (Throwable)null, 2, (Object)null);
   }

   this.thumbnailCache = (DiskLruCache)null;
   Unit var9 = Unit.INSTANCE;
}

```

Questions for self:
 - Why does the `finally` block stop the try block from being executed in the `synchronized` block?
 - The nested `finally` is also very confusing and seems super redundant. 🤔